### PR TITLE
topology2: Add HDA with DMIC, HDMI topologies support for ACE3 architecture

### PR DESCRIPTION
**Pull request by udaymb**
This entry creates a topology for the HDA codec with PDM0 DMIC for PTL.

### CODE REVIEW

File: tools/topology/topology2/production/tplg-targets-hda-generic.cmake [hunk 1]
```diff
@@ -63,6 +63,12 @@ PREPROCESS_PLUGINS=nhlt,NHLT_BIN=nhlt-sof-hda-generic-ace1-2ch.bin,\
 DMIC0_ENHANCED_CAPTURE=true,EFX_DMIC0_TDFB_PARAMS=line2_generic_pm10deg,\
 EFX_DMIC0_DRC_PARAMS=dmic_default"
 
+# HDA + HDMI + DMICs for PTL
+"sof-hda-generic\;sof-hda-generic-ace3-2ch\;PLATFORM=ptl,HDA_CONFIG=mix,NUM_DMICS=2,\
```
Comment (ujfalusi) [inline tools/topology/topology2/production/tplg-targets-hda-generic.cmake:67]: ACE3 should use the existing ace1 topology file if a DMIC blob is needed.
We don't have separate set of topologies for LNL (ACE2) either, the DMIC blob is the same.

@singalsu, can you clarify this?

**Review by ujfalusi — CHANGES_REQUESTED** [review]

**Review by ujfalusi — APPROVED** [review]
OK, ACE3 needs new DMIC blob after all...

**Comment by singalsu** [comment]
> OK, ACE3 needs new DMIC blob after all...

Yes it does, please add also build for ace2 tplg, they all are different HW IP.

**Review by singalsu — APPROVED** [review]
Thanks @udaymb !

**Comment by ujfalusi** [comment]
> > OK, ACE3 needs new DMIC blob after all...
> 
> Yes it does, please add also build for ace2 tplg, they all are different HW IP.

I tried to see the difference, but the blob is the same for all ACE platforms, the reserved bits/register handling is all done in zephyr driver, the blob parsing is identical. Also, @udaymb said that the ac1 blob works on PTL fine (which is not a surprise).

**Comment by singalsu** [comment]
> > > OK, ACE3 needs new DMIC blob after all...
> > 
> > 
> > Yes it does, please add also build for ace2 tplg, they all are different HW IP.
> 
> I tried to see the difference, but the blob is the same for all ACE platforms, the reserved bits/register handling is all done in zephyr driver, the blob parsing is identical. Also, @udaymb said that the ac1 blob works on PTL fine (which is not a surprise).

No, it's not same, see:

https://github.com/alsa-project/alsa-utils/blob/d45a2be9eaba2bb75840ca7c19bccc4d6578270b/topology/nhlt/intel/dmic/dmic-process.c#L790

**Review by bardliao — APPROVED** [review]

File: tools/topology/topology2/production/tplg-targets-hda-generic.cmake [hunk 2]
```diff
@@ -63,6 +63,12 @@ PREPROCESS_PLUGINS=nhlt,NHLT_BIN=nhlt-sof-hda-generic-ace1-2ch.bin,\
...
+PREPROCESS_PLUGINS=nhlt,NHLT_BIN=nhlt-sof-hda-generic-ace3-2ch.bin,\
+DMIC0_ENHANCED_CAPTURE=true,EFX_DMIC0_TDFB_PARAMS=line2_generic_pm10deg,\
+EFX_DMIC0_DRC_PARAMS=dmic_default"
+
```
Comment (ujfalusi) [inline tools/topology/topology2/production/tplg-targets-hda-generic.cmake:71]: Please add the other variants of the ACE3 HDA topologies also, not just the 2ch.

**Review by ujfalusi — CHANGES_REQUESTED** [review]

File: tools/topology/topology2/production/tplg-targets-hda-generic.cmake (hunk 2 above)
Comment (udaymb) [reply tools/topology/topology2/production/tplg-targets-hda-generic.cmake:85]: @ujfalusi - Updated multiple entries for creating ACE3 HDA

**Review by ujfalusi — CHANGES_REQUESTED** [review]

**Review by lgirdwood — APPROVED** [review]

**Comment by kv2019i** [comment]
SOFCI TEST

**Comment by kv2019i** [comment]
sof-docs fail and Intel LNL fails all known and tracked in https://github.com/thesofproject/sof/issues?q=is%3Aissue+is%3Aopen+label%3A%22Known+PR+Failures%22+

**Pull request merged**